site/en/blog/shared-element-transitions-for-spas/index.md

```diff
@@ -0,0 +1,196 @@
+
+### Shared element transitions
+
+Shared element transitions build on top of simple transitions by letting you specify elements to transition independently. This is how the header in the following example stays static during these transitions:
+
+{% Video
+  src="video/CZmpGM8Eo1dFe0KNhEO9SGO8Ok23/63bGGDsvP2sE7b009F6w.mp4",
+  loop=true,
+  muted=true,
+  playsinline=true,
+  controls=true
+%}
+
+But they don't need to stay static. They can also move independently of the root transition.
+
+Here's how it's done:
+
+```js
+// When the user clicks on a link/button:
+async function navigateToSettingsPage() {
+  // Capture and visually freeze the current state.
+  await document.documentTransition.prepare({
+    rootTransition: 'cover-up',
+    sharedElements: [element1, element2, element3],
+  });
+  // This is a function within the web app:
+  updateDOMForSettingsPage();
+  // Start the transition.
+  await document.documentTransition.start({
+    sharedElements: [element1, element4, element5],
+  });
+  // Transition complete!
+}
+```
+
+`sharedElements` is the difference here, both in the prepare step and the start step.
+
+In this case, `element1` will animate independently from the rest of the page, moving from its original position and size to its end position and size (or staying still if it doesn't move).
+
+The elements don't need to be the same node. In this case, `element2` will transition to `element4`, because they're each second in the provided array.
+
+#### Limitations of shared element transitions
+
+**When an element becomes 'shared', it shouldn't render on the background too, but right now it does**. This creates "ghosting" effects on elements that should appear static. This has been fixed, but hasn't made it to stable yet.
+
+**Shared transition elements will render on top of other elements.** If other elements render on top of the transitioned elements, they'll pop underneath for the transition. We're not sure yet if this should be a general limitation, or if it should be fixed.
+
+**Shared transition elements must have `contain: paint`.** This makes it difficult to transition elements that have paint effects outside their content rectangle, such as shadows, blurs, and outlines. Hopefully this will change in future.
+
+**Shared transition elements cannot have background effects.** Things like [`mix-blend-mode`](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode) and [`backdrop-filter`](https://developer.mozilla.org/en-US/docs/Web/CSS/backdrop-filter) won't interact correctly with the background during a transition.
+
+**Control over the transition is limited.** Right now, when one element transitions to another, their 'screenshots' quickly cross-fade while they scale and translate into position. Future versions of the API will add additional configuration, but this will always be limited to things that can be done with a screenshot of state, such as transforms, opacity, and crops.
```

Comment:
```suggestion
**Control over the transition is limited.** Right now, when one element transitions to another, their "screenshots" quickly cross-fade while they scale and translate into position. Future versions of the API will add additional configuration, but this will always be limited to things that can be done with a screenshot of state, such as transforms, opacity, and crops.
```